pymeasure/instruments/rigol/rigol_ds1000z.py

```
@@ -0,0 +1,917 @@
+    """Read and return the payload of a SCPI binary block (#Nxxxx...payload)."""
+    header = read_bytes_func(2)
+    if len(header) != 2 or not header.startswith(b"#"):
+        raise RuntimeError("Invalid binary block header for display data.")
+    length_digits = int(header[1:2].decode("ascii"))
+    length = int(read_bytes_func(length_digits).decode("ascii"))
+    payload = read_bytes_func(length)
+    # consume optional trailing terminator without failing if absent
+    try:
+        read_bytes_func(1, break_on_termchar=True)
+    except Exception:
+        pass
```

Ruff (0.15.10)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)